static/js/message_list.js

```diff
@@ -67,6 +67,16 @@ exports.MessageList.prototype = {
             // our current narrow to no longer be empty, hide the empty
             // feed placeholder text.
             narrow.hide_empty_narrow_message();
+            if (self.fetch_status.has_found_newest()) {
+                if (self.data._newest_items.length > 0) {
+                    const msg_list_data = self.data;
+                    let newest_items = msg_list_data._newest_items;
+                    newest_items = msg_list_data.valid_non_duplicated_messages(newest_items);
+                    newest_items = msg_list_data.append(newest_items);
+                    render_info = self.append_to_view(newest_items, opts);
+                    msg_list_data._newest_items = [];
+                }
+            }
```

Comment:
I'm not sure what the implications are of doing this additional `append_to_view`.  

Would it be easier to reason about if we just moved this logic in the `message_list_data.js` logic for `add_messages` there, and just had `if self.fetch_status.has_found_newest { messages.append(self._newest_items); self._newest_items = []; }`  there?

(I think we might need to move the `FetchStatus` object from the `MessageList` to the `MessageListData`, but that's probably an improvement in any case).

Reply:
> I'm not sure what the implications are of doing this additional `append_to_view`.

Will have to look into `append_to_view`, but without this I was getting the error:
```js
Error: Cannot read property 'top' of undefined
 at Object../static/js/message_viewport.js.exports.recenter_view /static/js/message_viewport.js:317
```

Also added a comment about `valid_non_duplicated_messages` after moving this logic.